openmicroscopy/openmicroscopy
# Open with right panel

# What this PR does

Adds support for Open-with to top of right panel.
See https://trello.com/c/Cdzu1KlJ/57-open-with-from-right-panel

This allows you to "Open-with" without using the tree right-click menu, particularly useful for Images in Wells, that you don't see in the tree.

# Testing this PR

1. In our current setup, we don't have any "open with" plugins that support any objects except Images.

2. Check that the Open_with button shows the list of open-with plugins (same as right-click on tree -> open with) and that they are enabled/disabled according to selected object(s) in the same way as in the tree.

3. Both disabled and enabled items should give an appropriate tooltip.

4. Clicking on enabled items should open link in new tab, or you can right-click on the item and choose browser option of "Open in new Window", as for any regular link.

5. Browse SPW -> Well -> Image. Check you can open image in iViewer or image(s) in figure.

6. Also, check that a new Open-with icon is used both in the tree menu and for the new button.

![screen shot 2017-04-04 at 11 11 42](https://cloud.githubusercontent.com/assets/900055/24652596/59c8ec06-192a-11e7-9f30-ef0de23934e5.png)



# Related reading

See https://trello.com/c/Cdzu1KlJ/57-open-with-from-right-panel

--depends-on #5285
